### DISCUSSION TITLE
FAQ / Knowledge Base

### DISCUSSION THREAD

**Discussion by leaanthony** [squidfunk/mkdocs-material#6174] [Q&A] [answered]
I was wondering if anyone knew of a solution for creating a searchable knowledge base / FAQ. I mean, you get search with the main site but I was looking for something like a plugin to make it more focused, eg a dedicated page with dedicated results. I guess I could IFrame a different site but having two sites seems overkill. Any thoughts?

**Comment by alexvoss** [comment]
Right now I am not sure if this can be done. One suggestion I have, though, is that you might want to consider [boosting](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/#search-boosting) the FAQ pages? 

In the longer term, I believe there are changes ahead for the search function that would help with this, so you might just need to sit tight to get a better solution than the one I can offer here. I am not sure what @squidfunk has on his todo list but being able to narrow a search by [tag](https://squidfunk.github.io/mkdocs-material/setup/setting-up-tags/) sounds like a useful feature...

**Reply by squidfunk** [reply]
You can already search by tag. If you tag your pages, tags will be included in search and can be searched.

**Reply by alexvoss** [reply]
Are they searched just like any word or is there a way to specifically narrow the search to results with that tag? Something like `tag: faq` in this case? Sorry if I have missed this - otherwise might be good documenting?

**Reply by squidfunk** [reply]
You should be able to use `tags:foo` to search for the tag `foo`. This is not documented, because we're not sure yet where we're heading with the new search, but searching on specific fields will likely be supported one way or the other.

**Reply by squidfunk** [reply]
By the way, we currently don't use tags on our own docs, because we haven't found a good taxonomy yet that makes sense to maintain aside from the regular site structure, but we'll definitely revisit it in the light of the new search implementation. I also want to add the ability to add tags to sections in the future, not only pages.

**Reply by leaanthony** [reply]
Thanks all for your time 🙏 I'll start tagging and see where I get.

**Comment by squidfunk** [comment]
@leaanthony Why do you want to use a dedicated site? What's the motivation?

**Reply by leaanthony** [reply]
Just to keep all the troubleshooting tips, etc separate to the general search results from the site. It'd be great to be able to filter the search by the knowledge base tag or whatever. 

What do other people do? Just have a section for an FAQ and use the standard search? The FAQ I'm looking to build is pretty big and has articles just for certain platforms so it would be nice to have something a bit more granular.

**Reply by squidfunk** [reply, marked as answer]
> Just to keep all the troubleshooting tips, etc separate to the general search results from the site. It'd be great to be able to filter the search by the knowledge base tag or whatever.

We're working on filtering capabilities. I'm currently focusing my work on the new search, and expect that it will be released later this year ☺️ Also related (and planned):

- https://github.com/squidfunk/mkdocs-material/issues/4983

**Reply by leaanthony** [reply]
That's awesome sir 👌🙏
